##############################################################################
# About
##############################################################################

# Basic idea is to use this to feed gopher behaviours with the correct
# names/topics for your robot so that you don't have to 1) hardcode everything
# or 2) do massive amounts of params/remaps in launchers. It also lets
# you conveniently standardise various ideas.

##############################################################################
# Names
##############################################################################
topics:
  display_notification: /gopher/status_notifier/display_notification
  initial_pose: /navi/initial_pose
  switch_map: /dslam/map/switch

services:
  clear_costmaps: /navi/move_base/clear_costmaps

namespaces:
  semantics: /rocon/semantics

frames:
  map: map


##############################################################################
# Interactions
##############################################################################

# Button aliases standardised for gopher deliveries,
buttons:
  go:   /gopher/buttons/go
  stop: /gopher/buttons/stop

# Sound aliases standardised for gopher deliveries,
# system wide aliases should be specified above. 
sounds:
  honk: /gopher/commands/sounds/groot

##############################################################################
# Patterns
##############################################################################
#
# LED Patterns are one set of configurations that can't yet be configured
# from yaml. These however are done in the interface:
#
#    python: gopher_configuration.configurations.Configuration
#    c++   : N/A
#
# For reference, *concepts* are:
#
#  - human interaction required : flashing
#  - notifications              : around
#  - states                     : solid
#
# and *core definitions*: 
#
# human interactions
#
#  - humans_give_me_input = gopher_std_msgs.Notification.FLASH_BLUE
#  - humans_be_careful = gopher_std_msgs.Notification.FLASH_YELLOW
#  - humans_i_need_help = gopher_std_msgs.Notification.FLASH_PURPLE
#
# notifications
#
#  - holding = gopher_std_msgs.Notification.AROUND_RIGHT_BLUE
#
# states
#
#  - error = gopher_std_msgs.Notification.SOLID_RED
